urllib3/connectionpool.py

```diff
@@ -666,6 +666,15 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            try:
+                # discard any remaining response body, the connection will be
+                # released back to the pool once the entire response is read
+                response.read()
+            except (TimeoutError, HTTPException, SocketError, ProtocolError,
+                    BaseSSLError, SSLError) as e:
+                pass
```

Comment:
We're going to need a test that covers this sensibly: that'll basically mean sending back a malformed HTTP response to trigger this error, or at least triggering a timeout.